### PR TITLE
Custom IP addres field

### PR DESCRIPTION
**Pull request by msazanov**
Added a field for entering the IP address manually

### CODE REVIEW

File: zwave_me_server/Dockerfile.aarch64 [hunk 1]
```diff
@@ -11,6 +11,8 @@ RUN apt-get update && \
 
 RUN wget -q -O - https://storage.z-wave.me/RaspbianInstall |bash
 RUN rm -f /opt/z-way-server/automation/storage/*
+RUN wget -O /opt/z-way-server/automation/classes/AutomationController.js https://raw.githubusercontent.com/Z-Wave-Me/home-automation/1c5ad7bb95453b5bb7749ad2d1e5907d6949ccc3/classes/AutomationController.js
+
```
Comment (PoltoS) [inline zwave_me_server/Dockerfile.aarch64:15]: Remove in the PR version

File: zwave_me_server/config.yaml [hunk 2]
```diff
@@ -18,15 +18,17 @@ options:
...
   remote_support_access: bool?
   zbw_password: str
+  local_ip: str
 webui: http://[HOST]:[PORT:8083]
 ports_description:
   8083/tcp: Z-Way UI
   8084/tcp: Webif UI
   8883/tcp: MQTT server port (use port 8883 in the MQTT app)
-image: "ghcr.io/z-wave-me/{arch}-ha-z-wave-me-addon"
+image: "ghcr.io/msazanov/{arch}-ha-z-wave-me-addon"
```
Comment (PoltoS) [inline zwave_me_server/config.yaml:34]: Revert in the PR version, keep in own branch

File: zwave_me_server/run.sh [hunk 3]
```diff
@@ -19,10 +19,11 @@ defCJ=/opt/z-way-server/automation/defaultConfigs/config.json
...
   device=$(grep -Eo '/dev/tty[A-Z]*[0-9]' /data/options.json)
-  forceDevice=$(grep "configjson_device_replace" /data/options.json | cut -d ":" -f 2 | tr -d ',')
-  remote_access=$(grep "remote_access" /data/options.json | cut -d ":" -f 2 | tr -d ',')
-  remote_support_access=$(grep "remote_support_access" /data/options.json | cut -d ":" -f 2 | tr -d ',')
-  zbw_password=$(grep "zbw_password" /data/options.json | cut -d ":" -f 2 | tr -d '," ')
+  forceDevice=$(grep -oP '(?<="configjson_device_replace": ).*(?=,)' /data/options.json)
+  remote_access=$(grep -oP '(?<="remote_access": ).*(?=,)' /data/options.json)
+  remote_support_access=$(grep -oP '(?<="remote_support_access": ).*(?=,)' /data/options.json)
+  zbw_password=$(grep -oP '(?<="zbw_password": ).*(?=,|\})' /data/options.json | tr -d '," ')
+  local_ip=$(grep -oP '(?<="local_ip": ").*(?=")' /data/options.json)
```
Comment (PoltoS) [inline zwave_me_server/run.sh:26]:
```suggestion
  forceDevice=$(jq -r '.configjson_device_replace | select(. != null)' /data/options.json)
  remote_access=$(jq -r '.remote_access | select(. != null)' /data/options.json)
  remote_support_access=$(jq -r '.remote_support_access | select(. != null)' /data/options.json)
  zbw_password=$(jq -r '.zbw_password | select(. != null)' /data/options.json)
  local_ip=$(jq -r '.local_ip | select(. != null)' /data/options.json)
```

File: zwave_me_server/run.sh [hunk 4]
```diff
@@ -92,10 +94,13 @@ else
   touch /etc/zbw/flags/no_connection
 fi
 
+##Little fix, remove this line when z-way 4.0.4 release
+wget -O /opt/z-way-server/automation/classes/AutomationController.js https://raw.githubusercontent.com/Z-Wave-Me/home-automation/1c5ad7bb95453b5bb7749ad2d1e5907d6949ccc3/classes/AutomationController.js
+sleep 5
```
Comment (PoltoS) [inline zwave_me_server/run.sh:99]: revert in the PR

File: zwave_me_server/run.sh [hunk 5]
```diff
@@ -92,10 +94,13 @@ else
...
 # Starting Z-Way services
 /etc/init.d/mongoose start
 /etc/init.d/z-way-server start
-/etc/init.d/zbw_connect start
+
```
Comment (PoltoS) [inline zwave_me_server/run.sh:103]: revert in the PR

File: zwave_me_server/run.sh (hunk 3 above)
Comment (PoltoS) [reply zwave_me_server/run.sh:26]: And add installation of jq in Dockerfile*

File: zwave_me_server/run.sh [hunk 6]
```diff
@@ -19,10 +19,11 @@ defCJ=/opt/z-way-server/automation/defaultConfigs/config.json
 if [ -f "/data/options.json" ]; then
   # Get device path from Configuration tab of addon
   device=$(grep -Eo '/dev/tty[A-Z]*[0-9]' /data/options.json)
```
Comment (PoltoS) [inline zwave_me_server/run.sh:21]:
```suggestion
  device=$(jq -r '.device | select(. != null)' /data/options.json)
```

**Review by PoltoS — CHANGES_REQUESTED** [review]

File: zwave_me_server/Dockerfile.aarch64 [hunk 7]
```diff
@@ -11,6 +11,8 @@ RUN apt-get update && \
...
 
 RUN useradd -rm -d /home/support -s /bin/bash -g root -G sudo -u 1000 support 
 RUN  echo 'support:razberry' | chpasswd
```
Comment (PoltoS) [inline zwave_me_server/Dockerfile.aarch64:18]:
```suggestion
RUN echo 'support:razberry' | chpasswd
```

File: zwave_me_server/run.sh [hunk 8]
```diff
@@ -92,6 +94,7 @@ else
   touch /etc/zbw/flags/no_connection
 fi
 
+
```
Comment (PoltoS) [inline zwave_me_server/run.sh:97]:
```suggestion
```

File: zwave_me_server/Dockerfile.armhf [hunk 9]
```diff
@@ -17,6 +17,9 @@ RUN  echo 'support:razberry' | chpasswd
 
 RUN echo "ha" > /etc/z-way/box_type
 
+RUN wget -O /opt/z-way-server/automation/classes/AutomationController.js https://raw.githubusercontent.com/Z-Wave-Me/home-automation/1c5ad7bb95453b5bb7749ad2d1e5907d6949ccc3/classes/AutomationController.js
+
+
```
Comment (PoltoS) [inline zwave_me_server/Dockerfile.armhf:22]:
```suggestion
```

File: zwave_me_server/Dockerfile.aarch64 [hunk 10]
```diff
@@ -17,6 +17,9 @@ RUN  echo 'support:razberry' | chpasswd
 
 RUN echo "ha" > /etc/z-way/box_type
 
+RUN wget -O /opt/z-way-server/automation/classes/AutomationController.js https://raw.githubusercontent.com/Z-Wave-Me/home-automation/1c5ad7bb95453b5bb7749ad2d1e5907d6949ccc3/classes/AutomationController.js
+
+
```
Comment (PoltoS) [inline zwave_me_server/Dockerfile.aarch64:22]:
```suggestion
```

File: zwave_me_server/Dockerfile [hunk 11]
```diff
@@ -20,6 +20,9 @@ RUN  echo 'support:razberry' | chpasswd
 
 RUN echo "ha" > /etc/z-way/box_type
 
+RUN wget -O /opt/z-way-server/automation/classes/AutomationController.js https://raw.githubusercontent.com/Z-Wave-Me/home-automation/1c5ad7bb95453b5bb7749ad2d1e5907d6949ccc3/classes/AutomationController.js
+
+
```
Comment (PoltoS) [inline zwave_me_server/Dockerfile:25]:
```suggestion
```

File: zwave_me_server/Dockerfile.armhf (hunk 9 above)
Comment (msazanov) [reply zwave_me_server/Dockerfile.armhf:22]: remove newline

File: zwave_me_server/Dockerfile.aarch64 (hunk 10 above)
Comment (msazanov) [reply zwave_me_server/Dockerfile.aarch64:22]: remove newline

**Pull request merged**